src/stan/variational/advi.hpp

```diff
@@ -285,26 +420,49 @@ namespace stan {
 
             if (print_stream_)
               *print_stream_ << std::endl;
+
+            if (do_more_iterations == false &&
+                rel_difference_(elbo, elbo_best) > 0.05) {
+              if (print_stream_)
+                *print_stream_
+                  << "Informational Message: The ELBO at a previous "
+                  << "iteration is larger than the ELBO upon "
+                  << "convergence!"
+                  << std::endl
+                  << "This means that the variational approximation has "
+                  << "not converged to the global optima."
+                  << std::endl;
+            }
           }
 
-          // Check for max iterations
-          if (iter_counter == max_iterations) {
+          if (iter_main == max_iterations) {
             if (print_stream_)
-              *print_stream_ << "MAX ITERATIONS REACHED" << std::endl;
+              *print_stream_
+                << "Informational Message: The maximum number of "
+                << "iterations is reached! The algorithm has not "
+                << "converged."
+                << std::endl
+                << "Values from this variational approximation are not "
+                << "guaranteed to be meaningful."
+                << std::endl;
             do_more_iterations = false;
           }
 
-          ++iter_counter;
+          ++iter_main;
```

Comment:
You can just make this whole thing a for loop and the structure will be clearer:

```
for (int iter_main = 1; do_more_iterations; ++iter_main) {
  ...
}
```

Two general principles:  keep variables defined in the scope in which they're used and use standard idioms like for loops when it's a for-loop structure.

We can go through and refactor some of this code later.  I don't want to postpone this merge indefinitely if the code's well tested.